File: Silver/2016/Dec_p2_seven_subsequences/div7.java
import java.io.*;
import java.util.*;

public class Div7 {
   public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new FileReader("div7.in"));

      int cn = Integer.parseInt(br.readLine());
      int[] cows = new int[cn];

      for (int i = 0; i < cn; i++) {
         cows[i] = Integer.parseInt(br.readLine());
      }
      
      br.close();

      List<Integer> l = new ArrayList<>();
      l.add(0);
      int longest = 0;

      for (int e : cows) {
         int val = (l.get(l.size() - 1) + e) % 7;
         l.add(val);
         int dist = l.size() - 1 - l.indexOf(val);
         if (dist > longest) {
               longest = dist;
         }
      }

      BufferedWriter bw = new BufferedWriter(new FileWriter("div7.out"));
      bw.write(Integer.toString(longest));
      bw.close();
   }
}

File: Silver/2016/Dec_p2_seven_subsequences/div7.py
f = open("div7.in", "r")


cn = int(f.readline())
cows = []
for _ in range(cn):
	cows.append(int(f.readline()))

l = [0]
longest = 0

for e in cows:
	val = (l[-1] + e) % 7
	l.append(val)
	dist = len(l) - 1 - l.index(val)
	if dist > longest:
		longest = dist


f = open("div7.out", "w")
f.write(str(longest))
f.close()